CHANGELOG.md

```diff
@@ -16,18 +16,51 @@
 
 ### Breaking changes
 
-* Lorem ipsum.
+* Disable copying of various classes to prevent incorrect use at compile time.
+* History type enumeration value `Replication::hist_Sync` renamed to
+  `Replication::hist_SyncClient`.
+* Bumps file format version from 6 to 7 due to addition of a 10th element into
+  `Group::m_top`. The new element is the history schema version, which is
+  crucial for managing the schema upgrade process of sync-type histories in a
+  way that is independent of core's Realm file format. The bump is necessary due
+  to lack of forwards compatibility. The changes are backwards compatible, and
+  automatic upgrade is implemented.
+* New pure virtual methods `get_history_schema_version()`,
+  `is_upgradable_history_schema()`, and `upgrade_history_schema()` in
+  `Replication` interface.
 
 ### Enhancements
 
 * Support setting Mixed(Timestamp) through the transaction logs.
 * Implement comparison of Mixed objects containing Timestamp types.
+* Allow query for size of strings, binaries, linklists and subtables:
+  Query q = table->where().size_equal(2, 5);
+  Query q = table1->column<SubTable>(2).size() == 5;
+* New history type enumeration value `Replication::hist_SyncServer`. This allows
+  for the sync server to start using the same kind of in-Realm history scheme as
+  is currently used by clients.
 
 -----------
 
 ### Internals
 
-* Lorem ipsum.
+* `StringIndex` now supports case insensitive searches.
+  PR [#2475](https://github.com/realm/realm-core/pull/2475).
+* `AppendBuffer` gained support for move construction/assignment, and had its
+  growth factor reduced to 1.5.
+* Methods on the `Replication` interface were made virtual to allow override.
+* The order of emission for some instructions in the transaction log was changed
+  with respect to carrying out the effect of the instruction on the database, to
+  allow implementors of the `Replication` interface a semi-consistent view of
+  the database.
+* Lock file format bumped from version 9 to 10 due to introduction of
+  `SharedInfo::history_schema_version`.
+* Removal of obsolete logic and semantics relating to obsolete history type
+  `Replication::hist_OutOfRealm`.
+* Code specific to history type `Replication::hist_InRealm` (class
+  `_impl::InRealmHistory` in particular) was moved from
+  `realm/impl/continuous_transactions_history.hpp` and
+  `realm/impl/continuous_transactions_history.cpp` to `realm/sync/history.cpp`.
```

Comment:
And to a lesser degree of necessity here. Still would be very nice, me thinks. 😄